ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/DecryptPioneerPayloads.java

```diff
@@ -115,21 +117,31 @@ private DecryptPioneerPayloads(ValueProvider<String> metadataLocation,
       validator.validate(envelopeSchema, json);
       JsonNode payload = json.get(FieldName.PAYLOAD);
 
-      String encryptionKeyId = payload.get(ENCRYPTION_KEY_ID).asText();
-      PrivateKey key = keyStore.getKey(encryptionKeyId);
-      if (key == null) {
-        // Is this really an IOException?
-        throw new IOException(String.format("encryptionKeyId not found: %s", encryptionKeyId));
-      }
-
-      final byte[] decrypted = decrypt(key, payload.get(ENCRYPTED_DATA).asText());
-
       byte[] payloadData;
-      if (decompressPayload.get()) {
-        payloadData = GzipUtil.maybeDecompress(decrypted);
+      if (payload.get(SCHEMA_NAME).asText().equals(DELETION_REQUEST_SCHEMA_NAME)) {
+        // The deletion-request ping is special cased within the decryption
+        // pipeline. The Pioneer client requires an encrypted payload in the
+        // payload, and the server (this transform in particular) requires the
+        // payload to contain encrypted data. We only require the injected
+        // metadata from the envelope to shred all data related to a client, so
+        // the encrypted data in the deletion-request ping is thrown away.
+        payloadData = "{}".getBytes(Charsets.UTF_8);
```

Comment:
Why do we need to clear the payload manually here?

Deletion-request schema specifies an empty document and would be used to validate these pings later on in the transformation chain. IIUC this guarantees an empty payload anyway.

Reply:
The client is required to encrypt something locally, which ends up being "" with a throwaway key. We need to replace the payload because of this.